### PR TITLE
GH WF to sync container repositories

### PR DESCRIPTION
**Pull request by snazy**
Adds a workflow to automatically synchronize our Docker container repositories to quay.io and ghcr.io. Includes all tags in `docker.io/projectnessie/nessie` + recent tags in `docker.io/projectnessie/nessie-unstable`.

This is a prerequisite part for #6293

### CODE REVIEW

File: .github/workflows/container-sync.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      - name: Sync images
+        env:
+          REGCLIENT_VERSION: 0.4.7
+          QUAY_IO_USER: ${{ secrets.QUAY_IO_USER }}
+          QUAY_IO_TOKEN: ${{ secrets.QUAY_IO_TOKEN }}
+          GHCR_IO_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          curl -L https://github.com/regclient/regclient/releases/download/v${REGCLINET_VERSION}/regsync-linux-amd64 > regsync
+          sha512sum -c .github/docker-sync/regsync.sha512
+          chmod 755 regsync
```
Comment (dimas-b) [inline .github/workflows/container-sync.yml:33]: Do we actually run it?

File: .github/workflows/container-sync.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  schedule:
+    # Run daily on week days
+    - cron:  '0 11,23 * * 1-5'
+  workflow_run:
+    workflows:
+      - 'Publish release'
+      - 'Publish in-development builds from main'
+    types:
+      - completed
+  workflow_dispatch:
```
[the remark this answers is not in the document]
Comment (snazy) [reply .github/workflows/container-sync.yml:13]: yes

[the remark this answers is not in the document]
Comment (snazy) [reply .github/workflows/container-sync.yml:13]: This allows the WF to be manually triggered

File: .github/workflows/container-sync.yml (hunk 1 above)
Comment (snazy) [reply .github/workflows/container-sync.yml:33]: :facepalm:

**Review by dimas-b — APPROVED** [review]

**Pull request merged**